persistence/sql/persister_identity.go

```diff
@@ -93,6 +93,42 @@ func (p *Persister) FindByCredentialsIdentifier(ctx context.Context, ct identity
 		IdentityID uuid.UUID `db:"identity_id"`
 	}
 
+	// #nosec G201
+	if err := p.GetConnection(ctx).RawQuery(fmt.Sprintf(`SELECT
+    ic.identity_id
+FROM %s ic
+    INNER JOIN %s ici on ic.id = ici.identity_credential_id
+WHERE ici.identifier = ?
+  AND ic.nid = ?
+  AND ici.nid = ?`,
+		"identity_credentials",
+		"identity_credential_identifiers",
+	),
+		match,
```

Comment:
Normalization was left out because OIDC credential identifier would be case-sensitive while Password/WebAuthn identifier's are lower-cased during identity creation. Prior to the query, we just have the identifier and not the credential type to perform the appropriate normalization

Reply:
I see, that's a fair assumption. Let's view it in the full context:

Only webauthn and password use user-based input for their identifiers (e.g. the email or username), which is why they get normalized (because users mess up capitalization on mobile). All other methods use either the identity's UUID or in the case of Social Sign In a concatenated string `provider:user_id_from_provider`

IMO we only need to cover the use case where the identifier is based on user input (aka registration or settings form - email, username, phone number, ...) because it doesn't really make sense to search for these system generated strings from an API perspective.

By the way, an addition to the logic would be to search for `recovery_addresses` or `verification_addresses` using the `via`. Not saying that we have to do it now, but can add later.